proto/substrait/algebra.proto

```diff
@@ -173,6 +173,8 @@ message JoinRel {
     // This join is useful for nested sub-queries where we need exactly one record in output (or throw exception)
     // See Section 3.2 of https://15721.courses.cs.cmu.edu/spring2018/papers/16-optimizer2/hyperjoins-btw2017.pdf
     JOIN_TYPE_SINGLE = 7;
+    JOIN_TYPE_MARK = 8;
```

Comment:
Do we want to add MARK join separately?  If so, that seems like the next logical step to get this change in.

Reply:
Fine with me; maybe the only thing we should take into consideration is the release dates.

The next DuckDB version is scheduled to release on the 2nd of September, and currently, the extension release dates are tightly bound to DuckDB's release date. I'm not sure if it would be more realistic to get just the mark join released in both Substrait and DuckDB by then, or if doing both in one go would be better, or if the release itself might not matter that much (e.g., if you use nightly builds or so).

Reply:
I'm working on getting MARK in again.  https://github.com/substrait-io/substrait/pull/682 is my latest attempt.  The revision to the way the mark join that Weston provided says that a column is being added (the mark column) -- is that actually happening outside of the join or is it merely an internal detail?

Reply:
As far as I understand the implementation of the Mark Join, an actual column of booleans is added to the join itself, so the explanation looks right to me.

@lnkuiper could you also double-check the description in the PR mentioned above regarding the mark-join?

Reply:
The description seems good to me. Good thinking adding both the left and right MARK join already. DuckDB currently only supports the left variant, but will probably add the right variant at some point in the future.